examples/large_models/mixtral_moe/hugging_face_llm_handler.py

```diff
@@ -0,0 +1,197 @@
+            for input_text in input_texts
+        ]
+        input_ids_batch, attention_mask_batch, params = [], [], []
+        logger.info(f"Batch size is {len(input_texts)}")
+        for input_text in input_texts:
+            logger.info(input_text)
+            input_ids, attention_mask = self.encode_input_text(input_text["prompt"])
+            input_ids_batch.append(input_ids)
+            attention_mask_batch.append(attention_mask)
+            params.append(input_text["params"])
+        input_ids_batch = torch.cat(input_ids_batch, dim=0).to(self.model.device)
+        attention_mask_batch = torch.cat(attention_mask_batch, dim=0).to(self.device)
+        return input_ids_batch, attention_mask_batch, params
+
+    def encode_input_text(self, input_text):
+        """
+        Encodes a single input text using the tokenizer.
+        Args:
+            input_text (str): The input text to be encoded.
+        Returns:
+            tuple: A tuple with two tensors: the encoded input ids and the attention mask.
+        """
+        inputs = self.tokenizer.encode_plus(
+            input_text,
+            max_length=self.max_length,
+            padding=True,
+            add_special_tokens=True,
+            return_tensors="pt",
+            truncation=True,
+        )
+        input_ids = inputs["input_ids"]
+        attention_mask = inputs["attention_mask"]
+        return input_ids, attention_mask
+
+    def inference(self, input_batch):
+        """
+        Predicts the class (or classes) of the received text using the serialized transformers
+        checkpoint.
+        Args:
+            input_batch (tuple): A tuple with two tensors: the batch of input ids and the batch
+                                of attention masks, as returned by the preprocess function.
+        Returns:
+            list: A list of strings with the predicted values for each input text in the batch.
+        """
+        input_ids_batch, attention_mask_batch, params = input_batch
+        logger.info(
+            f'temperature: {params[0]["temperature"]}, top_p: {params[0]["top_p"]}, max_new_tokens: {params[0]["max_new_tokens"]}'
+        )
+        input_ids_batch = input_ids_batch.to(self.device)
+
+        self.output_streamer = TextIteratorStreamerBatch(
+            self.tokenizer,
+            batch_size=len(input_ids_batch),
+            skip_special_tokens=True,
+        )
+        generation_kwargs = dict(
+            inputs=input_ids_batch,
+            attention_mask=attention_mask_batch,
+            streamer=self.output_streamer,
+            max_new_tokens=params[0]["max_new_tokens"],
+            temperature=params[0]["temperature"],
+            top_p=params[0]["top_p"],
+        )
+        thread = Thread(target=self.model.generate, kwargs=generation_kwargs)
+        thread.start()
+
+        for new_text in self.output_streamer:
+            send_intermediate_predict_response(
```

Comment:
you can check inf2 example to update this section to combine microbatch+streaming.